internal/workers/reportwkr/reportwkr.go

```diff
@@ -0,0 +1,50 @@
+package reportwkr
+
+import (
+	"context"
+	"github.com/davecgh/go-spew/spew"
+	"github.com/penguin-statistics/backend-next/internal/config"
+	"github.com/penguin-statistics/backend-next/internal/service"
+	"go.uber.org/fx"
+	"runtime"
+)
+
+type ReportConsumerDeps struct {
+	fx.In
+	ReportServices *service.Report
+}
+
+type ReportConsumerWorker struct {
+	// count is the number of workers
+	numWorker int
+
+	ReportConsumerDeps
+}
+
+func Start(conf *config.Config, deps ReportConsumerDeps) {
+	ch := make(chan error)
+	// handle & dump errors from workers
+	go func() {
+		for {
+			err := <-ch
+			spew.Dump(err)
+		}
+	}()
+	// works like a consumer factory
+	reportWorkers := &ReportConsumerWorker{
+		numWorker:          0,
+		ReportConsumerDeps: deps,
+	}
+	// spawn workers
+	// maybe we should specify the number of worker in config.Config ?
```

Comment:
This is indeed a better way to specify worker count. If you feel comfortable to do so you can implement this in this PR :D

Reply:
Can default `numWorker` be a fixed value? `Report.consumeReportTask()` seems not a CPU intensive task, therefore we might not want it to link together with the CPU count.

Reply:
> Can default `numWorker` be a fixed value? `Report.consumeReportTask()` seems not a CPU intensive task, therefore we might not want it to link together with the CPU count.

Indeed. Although the initial intention for this is to avoid spinning up too much workers on a low-end server like `t2.micro`, but that could just be fixed by add a config item instead of blindly using the CPU count.

Reply:
I think I could fix up this issue myself :D. Other than that, I'm gonna go ahead and merge this.